### ISSUE TITLE
15 Hour run/ Inventory overflow/ Ignored item limit/ Ignored min Cp and autotransfer.

### ISSUE THREAD

**Issue by Kegoscho** [jabbink/PokemonGoBot#494]
I ran the programm 15h or so and observed a strange behavior. 
For the first few hours (like 10 maybe) it was fine reached lvl 18 (this morning).
But after the last 5 hours i just gained 2 lvl. 
Also the bot ignored pokestops and wild Pokemon
Logged into the game. 

config.properties is the default  one.

Inventory 843/350

Item limit:
item_revive=20
item_max_revive=10
item_potion=0
item_super_potion=30
item_hyper_potion=50
item_max_potion=50
item_poke_ball=50
item_great_ball=50
item_ultra_ball=50
item_master_ball=50
item_razz_berry=50

instead it looked something like this
item_revive=90
item_max_revive=0
item_potion=110
item_super_potion=76
item_hyper_potion=140
item_max_potion=0
item_poke_ball=190
item_great_ball=80
item_ultra_ball=18
item_master_ball=0
item_razz_berry=130
 

My Poke bag:
many many  Pinsir/Golbat/Goldeen/Clefairy/Zubat
min_Cp was set to 400 but got many 50-300 CP in my bag 

**Version:**
Pokemon Go Bot v0.4.3

**Operating System:**
Windows 7

**Java version:**
java version "1.8.0_45"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)

**Comment by jabbink** [comment]
>Inventory 843/350 

?

**Comment by pashorizer** [comment]
@jabbink  Happens if you get level rewards. It exceeds the limit

**Comment by sodanakin** [comment]
If you want to be certain it doesn't hang for whatever reason in the middle of the night it might be a good idea to make a bat file to launch the Bot, and hook that bat file up to windows task scheduler and make it relaunch every hour. (You will also need a bat script to close the bot.) @Kegoscho 

I do this even while the bot runs perfectly for me in case of server issues.

**Comment by Kegoscho** [comment]
@jabbink got 843 Items in my 350 Inventory space =) 
@pashorizer ah ok thx 
@sodanakin ok I´ll give it a try.

**Comment by sodanakin** [comment]
@Kegoscho  Heres some examples http://puu.sh/qgiTe/37f990521f.zip . When making the task in windows task scheduler make sure to set the killscripts on top of the launcher so it will execute the kills first.

**Comment by Simpuhl** [comment]
@sodanakin 

The fact that you kill the bot at every hour and reopen it at every hour and you appear in the exact same place every hour thus causing a jump in GPS depending on where the bot was at when you closed it doesn't that make you feel like it's easy to trace and may get banned?

**Comment by Agronis** [comment]
@Simpuhl it's easy to trace regardless.

A simple query of "Show me the players who catch more than one pokemon in 12 seconds" will result in everyone's "ban"

**Comment by Simpuhl** [comment]
@Agronis 

That is true, I wish we could some how add delays in between catches and evolves etc.

**Comment by uk-shahj** [comment]
@sodanakin what process do you kill in the batch file "kill4"? Kill4 wasn't in the zip you uploaded so I was just wondering what process you killed in that one.

**Comment by ArtursGailis1995** [comment]
Due to some unstability issues with servers I would recommend adding option to auto-relaunch jarfile every 30 minutes or so (and killing previos one).

**Comment by Simpuhl** [comment]
@ArtursGailis1995 

i do not recommend this because  then your character would auto jump gps locations every 30 minutes.

**Comment by Kegoscho** [comment]
Correct me if  I´m wrong but this:

autotransfer=true                         //auto transfer duplicate pokemon
keep_pokemon_amount=1          //minimum amount of pokemon
transfer_cp_threshold=400
transfer_iv_threshold=80

should mean that after a while there should be every Pokemon (CP>400 and iv>80) im my Bag .. but only one of each and no duplicate Pokemon right ?

**Comment by michaeldajnowicz** [comment]
@simpuhl the bot only jumps around your set location.

**Comment by pashorizer** [comment]
`keep_pokemon_amount=0` for no duplicates

**Comment by Kegoscho** [comment]
@pashorizer  well XD thanks a lot

**Comment by Simpuhl** [comment]
Interesting, I thought 1 was to keep minimum of 1

**Comment by pashorizer** [comment]
Yeah the config is getting rich by now and some of the names are not really declarative...
With `keep_pokemon_amount=1` you will keep the highest rated one, but also have the possibility to get duplicates.

https://github.com/jabbink/PokemonGoBot/blob/develop/config.properties.template#L79-L80
There is also this option
`max_pokemon_amount=-1`

**Comment by michaeldajnowicz** [comment]
@Kegoscho also if you just re-run the script it should auto transfer/drop pokemons that the bot didn't pick up before.

**Comment by Kegoscho** [comment]
@michaeldajnowicz  yea .. but not all =S .. still got a lot of Zubats .... also many with CP<400 but still no transfer (keep is on 400)

and still after using the new config @pashorizer  it still keeps some pokemon duplicate ...

**Comment by Kegoscho** [comment]
Only if i put them into  obligatory_transfer ... but it ignores CP ... wait ... if 
transfer_iv_threshold=80
and 
transfer_cp_threshold=500 

will it still keep it when IV>80 but CP<500 ??? 
or will it only keep if both criteria are met?

**Comment by Letrico** [comment]
transfer_iv_threshold=80
transfer_cp_threshold=500

I have mentioned this behavior before that requested to toggle behavior.

pokemon1 300 iv 90 - KEEP (CP<400 AND IV >80)
pokemon2 cp999 iv 50 - KEEP (CP>500 AND IV < 90)
pokemon3 cp400 iv 50 - TRANSFER (CP<400 AND CP<80

**Comment by citruz** [comment]
I can confirm your issue. After having the bot run over night I have 451/350 items in my inventory.
I will have a look into it ;)

**Comment by citruz** [comment]
While having a look in the logs I found something interesting:
* Profile update says `Inventory 350/350` even though in the app its 451/350. 
* The dropUseLessItems Task didn't drop any items even though I had about 30 more potions than the limit
* After I restarted the bot the inventory values are correct again and items have been dropped

Seems like the inventory is not updated correctly even though `UpdateProfile` calls `inventories.updateInventories(true)` every 60 seconds. Maybe a bug in the API?
